sdmetrics/reports/single_table/_properties/column_shapes.py

```diff
@@ -0,0 +1,119 @@
+import warnings
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+
+from sdmetrics.reports.single_table._properties import BaseSingleTableProperty
+from sdmetrics.single_column import KSComplement, TVComplement
+
+
+class ColumnShapes(BaseSingleTableProperty):
+    """Column Shapes property class for single table.
+
+    This property assesses the shape similarity between the real and synthetic data.
+    A metric score is computed column-wise and the final score is the average over all columns.
+    The KSComplement metric is used for numerical and datetime columns while the TVComplement
+    is used for categorical and boolean columns.
+    The other column types are ignored by this property.
+    """
+
+    _sdtype_to_metric = {
+        'numerical': KSComplement,
+        'datetime': KSComplement,
+        'categorical': TVComplement,
+        'boolean': TVComplement
+    }
+
+    def _generate_details(self, real_data, synthetic_data, metadata, progress_bar=None):
+        """Generate the _details dataframe for the column shapes property.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data
+            synthetic_data (pandas.DataFrame):
+                The synthetic data
+            metadata (dict):
+                The metadata of the table
+            progress_bar (tqdm.tqdm or None):
+                The progress bar to use. Defaults to tqdm.
+        """
+        column_names, metric_names, scores = [], [], []
+        warning_messages = []
+        for column_name in metadata['columns']:
+            sdtype = metadata['columns'][column_name]['sdtype']
+            try:
+                if sdtype in self._sdtype_to_metric:
+                    metric = self._sdtype_to_metric[sdtype]
+                    column_score = metric.compute(
+                        real_data[column_name], synthetic_data[column_name]
+                    )
+                else:
+                    continue
+
+            except Exception as e:
+                column_score = np.nan
+                warning_messages.append(
+                        f"Unable to compute Column Shape for column '{column_name}'. "
+                        f'Encountered Error: {type(e).__name__} {e}'
+                )
+            finally:
+                if progress_bar:
+                    progress_bar.update()
+
+            column_names.append(column_name)
+            metric_names.append(metric.__name__)
+            scores.append(column_score)
+
+        if progress_bar:
+            progress_bar.close()
+
+        for message in warning_messages:
+            warnings.warn(message)
```

Comment:
Instead of raising the errors as a warning we might want to show them as a column in the details. I'm not positive but I believe that this is the current behavior